packages/core/src/actors/promise.ts

```diff
@@ -5,24 +5,42 @@ import {
   ActorRefFromLogic,
   AnyActorRef,
   EventObject,
+  MachineContext,
   NonReducibleUnknown,
-  Snapshot
+  Snapshot,
+  StateValue
 } from '../types.ts';
 
-export type PromiseSnapshot<TOutput, TInput> = Snapshot<TOutput> & {
-  input: TInput | undefined;
-};
+export interface PromiseState {
+  value?: StateValue;
+  context?: MachineContext;
+}
```

Comment:
it would be nice to add some type-level tests for usage with discriminated unions, including tests that would exercise `.getSnapshot()`